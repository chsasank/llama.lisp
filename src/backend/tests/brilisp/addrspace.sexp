;; CMD: guile ../../utils/sexp-json.scm < {filename} | python ../../brilisp.py | python ../../llvm.py | sed -e 's/alloca-[a-z]*/alloca/' -e 's/entry-[a-z]*/entry/'

(brilisp
    (define ((add5 void) (a (ptr float (addrspace 1))))
        (set (five float) (const 5.0))
        (store a five)
        (ret)))
