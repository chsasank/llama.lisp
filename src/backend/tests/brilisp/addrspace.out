; ModuleID = ""
target triple = "unknown-unknown-unknown"
target datalayout = ""

define void @"add5"(float addrspace(1)* %"a")
{
alloca:
  %"a.1" = alloca float addrspace(1)*
  %"five" = alloca float
  br label %"entry"
entry:
  store float addrspace(1)* %"a", float addrspace(1)** %"a.1"
  store float 0x4014000000000000, float* %"five"
  %".5" = load float addrspace(1)*, float addrspace(1)** %"a.1"
  %".6" = load float, float* %"five"
  store float %".6", float addrspace(1)* %".5"
  ret void
}

